src/core/CPU.cpp

```diff
@@ -817,12 +820,24 @@ Blaze::Cycles Blaze::CPU::executePEA() {
 };
 
 Blaze::Cycles Blaze::CPU::executePEI() {
-	// TODO
+    Word address = load16(PC + 1);
+    Word effectiveAddress = load16(address);
+    store16(0, SP + 1, effectiveAddress);
+    //SP -= 2;
 	return 0;
 };
 
 Blaze::Cycles Blaze::CPU::executePER() {
-	// TODO
+    Word address = load16(PC + 1);
+    Word relativeAddr = PC + address;
```

Comment:
Same as above: you can let `decodeAddress` do this for you:
```suggestion
    Word relativeAddr = decodeAddress(AddressingMode::ProgramCounterRelativeLong);
```